src/app/shopping-list/shopping-list/shopping-list.component.ts

```diff
@@ -155,13 +163,16 @@ export class ShoppingListComponent implements OnInit, AfterViewInit, OnDestroy {
 		}
 		if (this.mode == 'edit') {
 			this.mode = 'add';
-			if (!this.editingIngredientId) return ngForm.resetForm();
+			if (!this.editingIngredient?.id) return ngForm.resetForm();
 			const updatedIngredient: IngredientsInStore = {
-				id: this.editingIngredientId,
+				id: this.editingIngredient?.id,
 				...ingredient,
 			};
 			this.store.dispatch(
-				shoppingListActions.updateIngredient({ ingredient: updatedIngredient })
+				shoppingListActions.updateIngredient({
+					ingredient: updatedIngredient,
+					previewIngredient: this.editingIngredient,
+				})
```

Comment:
This is very confusing and unclear
You are sending in the same action updated Ingredient and edited Ingredient (what's the difference between those?) and ingredient and preview Ingredient? What is preview ingredient and why it's preview?

Reply:
Sorry, I should be 'previous' not 'preview'. Pervious value is use for grouping logic in effect.

Reply:
So please rename it to previous :)